### PR TITLE
Fix hash_equals() php warning

### PR DESCRIPTION
**Pull request by davidsword**
This solves #977

`hash_equals()` expects both values passed into it to be strings. However `get_proxy_verification_key()` may return an integer when `WPCOM_VIP_PROXY_VERIFICATION` is undefined, triggering a PHP warning. Casting `rand()` to a  string prevents this.

phpunit tests updated to reflect string instead of integer. Added `assertNotEmpty()` to verify that the string of numbers are properly returned when `WPCOM_VIP_PROXY_VERIFICATION` undefined or blank.

### CODE REVIEW

File: lib/proxy/ip-forward.php [hunk 1]
```diff
@@ -152,15 +152,15 @@ function set_remote_address( $ip ) {
 /**
  * Return the defined verification key for a site
  *
- * @return string|int The verification key if available, or a random integer if no key is configured.
+ * @return string The verification key if available, or a string of random numbers if no key is configured.
  */
 function get_proxy_verification_key() {
 	if ( defined( 'WPCOM_VIP_PROXY_VERIFICATION' ) && ! empty( WPCOM_VIP_PROXY_VERIFICATION ) ) {
 		return WPCOM_VIP_PROXY_VERIFICATION;
```
Comment (mjangda) [inline lib/proxy/ip-forward.php]: Just to be safe, we should probably cast this as a string as well and add a test for it.

File: tests/test-lib-proxy-ip-forward.php [hunk 2]
```diff
@@ -226,7 +226,8 @@ public function test__not_defined() {
 
 		$actual_key = get_proxy_verification_key();
 
-		$this->assertTrue( is_numeric( $actual_key ) );
+		$this->assertNotEmpty( $actual_key );
+		$this->assertTrue( is_string( $actual_key ) );
```
Comment (mjangda) [inline tests/test-lib-proxy-ip-forward.php]: Any time you have multiple assertions in a test, it's helpful to add the message parameter with details about the failure.

**Comment by davidsword** [comment]
Thanks! Very good points. I've added a commit https://github.com/Automattic/vip-go-mu-plugins/pull/980/commits/7c9a0e0eb36096c90670fedfa5f3619ed2d1179e with these changes.

**Comment by mjangda** [comment]
r119769-deploy

**Pull request merged**